src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,8 +1,42 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private final DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
 
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("Report for period ").append(dateFrom)
+                .append(" - ").append(dateTo).append("\n");
+        LocalDate dateFromLocalTime = LocalDate.parse(dateFrom, dateTimeFormatter);
+        LocalDate dateToLocalTime = LocalDate.parse(dateTo, dateTimeFormatter);
+        LocalDate currentEmployeeDate;
+        String[] currentEmployee;
+        int hoursWorked;
+        int moneyPerHour;
+        int totalIncome;
+
+        for (String name : names) {
+            totalIncome = 0;
+            for (String datum : data) {
+                if (datum.contains(name)) {
+                    currentEmployee = datum.split(" ");
+                    currentEmployeeDate = LocalDate.parse(currentEmployee[0],
+                            dateTimeFormatter);
+                    if ((currentEmployeeDate.isAfter(dateFromLocalTime)
+                            && currentEmployeeDate.isBefore(dateToLocalTime))
+                            || currentEmployeeDate.equals(dateToLocalTime)) {
```

Comment:
what if `currentEmployeeDate.equals(dateFromLocalTime)` ?

Reply:
It was with currentEmployeeDate.equals(dateFromLocalTime), I wanted to test if it would work without it, I'll return it)